src/App.tsx

```diff
@@ -1,26 +1,38 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { Route, Switch, useHistory } from 'react-router-dom'
+import { useDispatch, useSelector } from 'react-redux'
+import { Home, SignIn, TypingTest, Scorecard } from './pages'
+import { AppBar } from './components'
+import {
+  HOME_ROUTE,
+  SIGN_IN_ROUTE,
+  TYPING_TEST_ROUTE,
+  SCORECARD_ROUTE,
+} from './constants'
+import { userLogout } from './redux'
+import { PrivateRoute } from './routes'
+import './App.css'
 
 function App() {
+  const dispatch = useDispatch()
+  const userLogin = useSelector((state: any) => state.userLogin)
+  const { refreshToken = '' } = (userLogin && userLogin?.userInfo) || ''
```

Comment:
Think we don't need to check if `userLogin` is truthy if we are  using optional chaining. e.g. it can be `const { refreshToken = '' } = userLogin?.userInfo || ''`